src/darwin/Framework/CHIP/CHIPCluster.h

```diff
@@ -80,6 +95,21 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (strong, nonatomic, nullable) NSNumber * fabricFiltered;
 
+/**
+ * Sets the data version for the Read Request for the interaction.
+ *
+ * This tag can be omitted if the value of EnableTagCompression in the Path field is true.
+ * In this case, the value for the omitted tag SHALL be set to the value for that
+ * tag (if present) in the last AttributeDataIB that had tag compression disabled
+ * and was seen in a message that is part of the same interaction model Action
+ * as the current message.
+ *
+ * If this tag was not present and tag compression was disabled,
+ * it SHALL be interpreted as though a data version was not specified in that,
+ * or subsequent AttributeDataIBs.
```

Comment:
This does not make any sense, again.

But also, if the data version is not nil and matches the server-side data version... what is the actual behavior of the read?  I suspect it doesn't really work very well.  How was this part tested?

Reply:
I am not certain what the server side version is, but I can look into it and attempt it. I tried with version 0 and 1, the read was successful. Although with write, I kept getting incompatible version error from the server.

Reply:
Updated comment.

Reply:
Ok. I removed data version from read attribute and created an issue to add it back is OnDone callback is added [here](https://github.com/project-chip/connectedhomeip/issues/18339).